AppKit/CPWindow/CPWindow.j

```diff
@@ -3844,6 +4069,46 @@ var interpolate = function(fromValue, toValue, progress)
 
 @end
 
+@implementation CPWindow (TrackingArea)
+
+- (void)_addToViewsWithTrackingAreas:(CPView)aView
+{
+    [_viewsWithTrackingAreas addObject:aView];
+    
+    var tas = [aView trackingAreas];
+    
+    for (var i = 0, nb = [tas count]; i < nb; i++)
+        [self _addTrackingArea:[tas objectAtIndex:i]];
```

Comment:
OK, I agree on the simplification in the addTrackingArea and logic is respected using the _addTrackingAreaView on the _setWindow. I can go this way.